flashlib.core.robot/src/main/java/com/flash3388/flashlib/robot/control/PidController.java

```diff
@@ -173,32 +140,35 @@ public void setOutputLimit(double outputLimit){
      * Calculates the output to the system to compensate for the error.
      *
      * @param processVariable the process variable of the system.
-     * @param setPoint the desired set point.
+     * @param setpoint the desired set point.
      *
      * @return the compensation value from the PID loop calculation
      */
-    public double calculate(double processVariable, double setPoint){
+    public double applyAsDouble(double processVariable, double setpoint, Time currentTimeSeconds) {
         if(mSetPointRange != 0) {
             processVariable = ExtendedMath.constrain(processVariable, processVariable - mSetPointRange, processVariable + mSetPointRange);
         }
 
-        double error = setPoint - processVariable;
+        double error = setpoint - processVariable;
 
         double pOut = mKp.getAsDouble() * error;
-        double fOut = mKf.getAsDouble() * setPoint;
+        double fOut = mKf.getAsDouble() * setpoint;
 
         if(mIsFirstRun){
             mIsFirstRun = false;
+            mTotalError = 0;
             mLastOutput = pOut + fOut;
-            mLastProcessVariable = processVariable;
+            mLastError = error;
+            mLastTimeSeconds = currentTimeSeconds.sub(Time.seconds(1));
```

Comment:
Hmmmmmm why are you subbing with `1`? For all we know `currentTimeSeconds` is less than 1 seconds.

Reply:
The first time result will contain error-lastError (where lastError = error) so: 0/timeDelta (so time delta could be anything besides 0) but I am removing time anyway